### PR TITLE
[Dask] Disable watch

### PR DESCRIPTION
**Pull request by AlonMaor14**
Since Dask runtime executes the handler / code locally, there is no point to watch the run logs as it blocks the client until completion either way.
https://jira.iguazeng.com/browse/ML-3296
And some deprecation work as described in https://jira.iguazeng.com/browse/ML-3104

### CODE REVIEW

**Review by liranbg — COMMENTED** [review]
See comments below

File: mlrun/runtimes/daskjob.py [hunk 1]
```diff
@@ -408,10 +408,8 @@ def deploy(
         )
 
     def gpus(self, gpus, gpu_type="nvidia.com/gpu"):
-        # TODO: In 1.4.0 change to NotImplementedError
-        raise DeprecationWarning(
-            "Dask's gpus is deprecated and will be removed in 1.4.0, use "
-            "with_scheduler_limits/with_worker_limits instead",
+        raise NotImplementedError(
```
[the remark this answers is not in the document]
Comment (AlonMaor14) [reply mlrun/runtimes/daskjob.py:411]: I mistakenly decided to remove in 1.4.0 and later found the ticket in the description with the deprecation guidelines which says it should be removed now

File: mlrun/runtimes/base.py [hunk 2]
```diff
@@ -466,7 +466,7 @@ def run(
             # single run
             try:
                 resp = self._run(run, execution)
-                if watch and self.kind not in ["", "handler", "local"]:
+                if watch and self.kind not in ["", "handler", "local", "dask"]:
```
[the remark this answers is not in the document]
Comment (guy1992l) [reply mlrun/runtimes/base.py:469]: Just a thought: I think there should be an enum for `RunTimeKinds`. There you can write methods like `def is_watchable(self) -> bool` and more to not have to edit lists in multiple locations of code. 

Also, what is `kind=""`?

**Review by guy1992l — APPROVED** [review]
Why the `NotImplemmentedError` is used instead of simply deleting the methods? Just to give the message?

If so, I would use the `with_scheduler/worker_requests` from these methods and print the user warning. In my eyes it seems healthier. Next version we may delete them all at once.

**Comment by AlonMaor14** [comment]
> Why the `NotImplemmentedError` is used instead of simply deleting the methods? Just to give the message?
> 
> If so, I would use the `with_scheduler/worker_requests` from these methods and print the user warning. In my eyes it seems healthier. Next version we may delete them all at once.

Actually, it's to override the base class methods instead of removing them as it was described in the deprecation ticket https://jira.iguazeng.com/browse/ML-3104

File: mlrun/runtimes/base.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (AlonMaor14) [reply mlrun/runtimes/base.py:469]: @guy1992l I agree and this came to my mind as well
empty kind is a special case of `local` I don't exactly know when / why it happens but it's there

[the remark this answers is not in the document]
Comment (AlonMaor14) [reply mlrun/runtimes/base.py:469]: @guy1992l seems like this is already a thing `mlrun.runtimes.RuntimeKinds`

**Review by Tankilevitch — CHANGES_REQUESTED** [review]
very nice job, I would add test to make sure we really skip the watch and that job completes

File: tests/run/test_run.py [hunk 3]
```diff
@@ -135,12 +136,51 @@ def test_local_runtime_hyper():
...
+    ],
+)
+def test_is_watchable(rundb_mock, kind, watch, expected_watch_count):
+    mlrun.RunObject.logs = Mock()
+    spec = tag_test(base_spec, "test_local_runtime")
+    func = new_function(
+        command=f"{examples_path}/handler.py",
+        kind=kind,
+    )
+    func.spec.remote = False
```
Comment (Tankilevitch) [inline tests/run/test_run.py:170]: why you need to set this?

**Review by Tankilevitch — CHANGES_REQUESTED** [review]

**Review by liranbg — APPROVED** [review]

File: tests/run/test_run.py (hunk 3 above)
Comment (AlonMaor14) [reply tests/run/test_run.py]: added comments

**Review by Tankilevitch — APPROVED** [review]
LGTM

**Pull request merged**